src/org/gk/qualityCheck/HumanReactionsWithNonHumanComplexesWithHumanComponentsCheck.java

```diff
@@ -0,0 +1,122 @@
+            if (!QACheckUtilities.isHumanDatabaseObject(physicalEntity)
+                    && physicalEntity.getSchemClass().isa(ReactomeJavaConstants.Complex)) {
+
+                // Find any human Components in the non-human Complex.
+                nonHumanComplexesWithHumanComponentsMap.put(physicalEntity, findAllHumanComponentsInComplex(physicalEntity));
+            }
+        }
+        return nonHumanComplexesWithHumanComponentsMap;
+    }
+
+    /**
+     * Finds any human Components in the incoming Complex.
+     * @param complex GKInstance -- Complex with non-human species or no species.
+     * @return Set<GKInstance> -- Any Components in incoming Complex with human species.
+     * @throws Exception -- Thrown by MySQLAdaptor.
+     */
+    private Set<GKInstance> findAllHumanComponentsInComplex(GKInstance complex) throws Exception {
+        // Only find GKInstances within incoming Complex. It is recursive, so if Complex-within-Complex, it will return ALL components.
+        Set<GKInstance> humanPEs = new HashSet<>();
+        for (GKInstance physicalEntity : QACheckUtilities.findAllConstituentPEs(complex)) {
+            if (QACheckUtilities.isHumanDatabaseObject(physicalEntity)) {
+                humanPEs.add(physicalEntity);
+            }
+        }
+        return humanPEs;
+    }
+
+    private String getReportLine(GKInstance event, GKInstance complex, GKInstance component) throws Exception {
+        String complexCreatedName = QACheckUtilities.getInstanceAttributeName(complex, ReactomeJavaConstants.created);
+        String componentCreatedName = QACheckUtilities.getInstanceAttributeName(component, ReactomeJavaConstants.created);
+        return String.join("\t",
+                event.getDBID().toString(),
+                event.getDisplayName(),
+                complex.getDBID().toString(),
+                complex.getDisplayName(),
+                component.getDBID().toString(),
+                component.getDisplayName(),
+                component.getSchemClass().getName(),
+                complexCreatedName,
+                componentCreatedName);
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Complexes_That_Should_Have_Homo_sapiens_Species";
+    }
+
+    private String[] getColumnHeaders() {
+        return new String[]{"DB_ID_RlE", "DisplayName_RlE", "DB_ID_Complex", "DisplayName_Complex", "DB_ID_Component", "DisplayName_Component", "Class_Component", "Created_Complex", "Created_Component"};
+    }
+
+
+    // Unused, but required, AbstractQualityCheck methods.
+    @Override
+    public void check() {
+    }
+
+    @Override
+    public void check(GKInstance instance) {
+    }
+
+    @Override
+    public void check(List<GKInstance> instances) {
+    }
+
+    @Override
+    public void check(GKSchemaClass cls) {
+    }
+
+    @Override
+    public void checkProject(GKInstance event) {
+    }
+
+    @Override
+    protected InstanceListPane getDisplayedList() { return null; }
```

Comment:
It's a little strange to extend a class but not use the required inherited method.  Does this class have to extend from "AbstractQualityCheck"?

Reply:
Yes it was strange. It was a (not unprecedented) attempt to keep the structure of the QA checks in line with they were built for this project. This code is being moved to release-qa now though, so there won't be any need for these unused methods.